utils/app/clean.ts

```diff
@@ -82,8 +82,9 @@ export const cleanConversationHistory = (history: any[]): Conversation[] => {
         conversation.folderId = null;
       }
 
-      if (!conversation.messages) {
-        conversation.messages = [];
+      if (!conversation.messages || conversation.messages.length === 0) {
+        // if there are no messages, we don't want to add the conversation to the history
```

Comment:
This might be a little controversial since I'm permanently deleting people's data if this gets deployed and maybe am not aware of some workflows that need this but if there aren't any messages I wanted to keep the conversation history clean. I think OpenAI's UI doesn't save chats in its sidebar that don't have any messages in them so this matches that behavior.